### PR TITLE
TW-1454: fix list search messages empty when load more

### PR DESCRIPTION
**Pull request by sherlockvn**
### Demo mobile:

Issue: #1454

https://github.com/linagora/twake-on-matrix/assets/43041967/54850388-e3b0-4d40-9fb0-d8327f944e5d

### CODE REVIEW

File: lib/domain/app_state/search/server_search_state.dart [hunk 1]
```diff
@@ -35,3 +35,10 @@ class ServerSearchChatFailed extends Failure {
   @override
   List<Object?> get props => [exception];
 }
+
+class ServerSearchLoading extends Success {
```
Comment (nqhhdev) [inline lib/domain/app_state/search/server_search_state.dart:39]: IMO: I don't see you using it in `Interactor`, where to use it?

File: lib/pages/search/server_search_controller.dart [hunk 2]
```diff
@@ -74,7 +74,14 @@ class ServerSearchController with SearchDebouncerMixin {
       (success) {
         if (success is ServerSearchChatSuccess) {
           updateNextBatch(success.nextBatch);
-          if (success.results?.isEmpty == true) {
+          List<Result>? searchResults;
+          if (searchResultsNotifier.value is PresentationServerSideSearch) {
```
Comment (nqhhdev) [inline lib/pages/search/server_search_controller.dart:78]: IMO: Why do I need this condition?

Comment (sherlockvn) [reply lib/pages/search/server_search_controller.dart:78]: for cast safety

File: lib/domain/app_state/search/server_search_state.dart (hunk 1 above)
Comment (sherlockvn) [reply lib/domain/app_state/search/server_search_state.dart:39]: okey, change in interactor from ServerSearchInitial => ServerSearchLoading

**Pull request closed without merging**